src/EnergyPlus/DataGlobalConstants.cc

```diff
@@ -652,6 +659,112 @@ namespace DataGlobalConstants {
         return ResourceTypeChar;
     }
 
+    enum FuelTypeNumber
+    {
+        ELECTRICITY,
+        NATURALGAS,
+        DIESEL,
+        GASOLINE,
+        COAL,
+        FUELOILNO1,
+        FUELOILNO2,
+        PROPANE,
+        OTHERFUEL1,
+        OTHERFUEL2
+    };
+
+    FuelTypeNumber ConvertFueltypeString(std::string FuelTypeInput)
+    {
+        if (FuelTypeInput == "ELECTRICITY")
+            return ELECTRICITY;
+        else if (FuelTypeInput == "NATURALGAS")
+            return NATURALGAS;
+        else if (FuelTypeInput == "DIESEL")
+            return DIESEL;
+        else if (FuelTypeInput == "GASOLINE")
+            return GASOLINE;
+        else if (FuelTypeInput == "COAL")
+            return COAL;
+        else if (FuelTypeInput == "FUELOILNO1")
+            return FUELOILNO1;
+        else if (FuelTypeInput == "FUELOILNO2")
+            return FUELOILNO2;
+        else if (FuelTypeInput == "PROPANE")
+            return PROPANE;
+        else if (FuelTypeInput == "OTHERFUEL1")
+            return OTHERFUEL1;
+        else if (FuelTypeInput == "OTHERFUEL2")
+            return OTHERFUEL2;
+    }
+
+    void ValidateFuelType(std::string FuelTypeInput)
+    {
+        // FUNCTION INFORMATION:
+        //       AUTHOR         Dareum Nam
+        //       DATE WRITTEN   May 2020
+        //       MODIFIED       na
+        //       RE-ENGINEERED  na
+
+        // PURPOSE OF THIS FUNCTION:
+        // Validates fuel types and sets output strings
+
+        FuelTypeNumber Fuel = ConvertFueltypeString(FuelTypeInput);
+
+        switch (Fuel) {
+        case ELECTRICITY:
+            FuelType = "Electric";
+            FuelTypeNumforAssignResource = DataGlobalConstants::AssignResourceTypeNum("ELECTRICITY");
+            FuelTypeNum = 1; // FuelTypeElectricity in DXCoils.cc and HVACVariableRefrigerantFlow.cc
+            break;
```

Comment:
So is setting 3 vars efficient? Is FuelTypeNum only used in DXCoils and HVACVariableRefrigeratntFlow? Seems like you need only 1 int here. Is FuelTypeNumforAssignResource and FuelTypeNum the same value? Seems likely.

Reply:
FuelTypeNum is used in DXcoils and HVACVariantRefrigerantFlow, and FuelTypeNumforAssignResouce is used in Boilers and Boilersteam with AssignResourceTypeNum() function (https://github.com/NREL/EnergyPlus/blob/develop/src/EnergyPlus/DataGlobalConstants.cc#L325-L490), but these two are not the same int value for each fuel type, so once I separated these two int vars.